### PR TITLE
Use rbe_configs_gen utility for generating RBE config

### PR DESCRIPTION
**Pull request by yanavlasov**
Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

**Comment by yanavlasov** [comment]
Incomplete, windows changes are still to be done

**Comment by yanavlasov** [comment]
@smukherj1

File: WORKSPACE [hunk 1]
```diff
@@ -4,9 +4,9 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
 http_archive(
     name = "bazel_toolchains",
-    sha256 = "4fb3ceea08101ec41208e3df9e56ec72b69f3d11c56629d6477c0ff88d711cf7",
-    strip_prefix = "bazel-toolchains-3.6.0",
-    urls = ["https://github.com/bazelbuild/bazel-toolchains/releases/download/3.6.0/bazel-toolchains-3.6.0.tar.gz"],
+    sha256 = "1adf7a8e9901287c644dcf9ca08dd8d67a69df94bedbd57a841490a84dc1e9ed",
+    strip_prefix = "bazel-toolchains-5.0.0",
+    urls = ["https://github.com/bazelbuild/bazel-toolchains/archive/v5.0.0.tar.gz"],
```
Comment (smukherj1) [inline WORKSPACE:9]: Replace with
```
    sha256 = "5b6eef36e1f627bf739d3212487c7c31e9f19ce68bcb4d672aa11be1bd6bd2ee",
    strip_prefix = "bazel-toolchains-5.1.0",
    urls = ["https://github.com/bazelbuild/bazel-toolchains/archive/v5.1.0.tar.gz"],
```

v5.1.0 contains some fixes for Windows

File: toolchains/regenerate.sh [hunk 2]
```diff
@@ -17,35 +17,40 @@ if [[ "true" == "${COMMIT_TOOLCHAINS}" ]]; then
...
 
 case ${OS_FAMILY} in
   linux)
-    RBE_BAZEL_TARGET_LIST="@rbe_ubuntu_clang_gen//... @rbe_ubuntu_clang_libcxx_gen//... @rbe_ubuntu_gcc_gen//..."
+    TOOLCHAIN_LIST="clang clang_libcxx gcc"
+    BAZELRC_TEMPLATE=${RBE_AUTOCONF_ROOT}/toolchains/linux.latest.bazelrc
     ;;
   windows)
-    RBE_BAZEL_TARGET_LIST="@rbe_windows_msvc_cl_gen//... @rbe_windows_clang_cl_gen//..."
+    TOOLCHAIN_LIST="@rbe_windows_msvc_cl_gen//... @rbe_windows_clang_cl_gen//..."
```
[the remark this answers is not in the document]
Comment (yanavlasov) [reply toolchains/regenerate.sh]: done

File: WORKSPACE (hunk 1 above)
Comment (yanavlasov) [reply WORKSPACE]: done

File: WORKSPACE [hunk 3]
```diff
@@ -16,6 +16,6 @@ load(
 
 bazel_toolchains_repositories()
 
-load("//toolchains:rbe_toolchains_config.bzl", "rbe_toolchains_config")
+#load("//toolchains:rbe_toolchains_config.bzl", "rbe_toolchains_config")
```
Comment (sunjayBhatia) [inline WORKSPACE:19]: nit: seems like this and the below line is removable?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -86,6 +97,14 @@ jobs:
   pool:
     vmImage: 'windows-latest'
   steps:
+  - bash: |
+      curl -fsSL --output "$TEMP/go.zip" "$GO_ARCHIVE_URL"
+      echo "$GO_ARCHIVE_CHECKSUM $TEMP/go.zip" | sha256sum --check
+      unzip "$TEMP/go.zip" -d "/usr/bin/go.1.16"
+      ln -s /usr/bin/go.1.16/bin/go /usr/bin/go
```
Comment (sunjayBhatia) [inline azure-pipelines.yml:104]: Seems like we're running into git bash symlinking issues

Could be worth it to just do what we did here: https://github.com/envoyproxy/envoy-build-tools/blob/9f058c4f2b24de42da5e965f878d4ded865a1be4/azure-pipelines.yml#L69 (add the relevant directory to the path) or use `cmd` to make the symlink, or try messing with the git bash configs and setting `winsymlinks:nativestrict`

See:
- https://www.joshkel.com/2018/01/18/symlinks-in-windows/
- https://www.cygwin.com/cygwin-ug-net/using.html#pathnames-symlinks

File: azure-pipelines.yml [hunk 5]
```diff
@@ -86,10 +96,19 @@ jobs:
...
+      GO_ARCHIVE_CHECKSUM: 5cc88fa506b3d5c453c54c3ea218fc8dd05d7362ae1de15bb67986b72089ce93
+    displayName: Install go 1.16
   - template: .azure-pipelines/generate_toolchains.yml
     parameters:
       bazeliskInstallCommandPrefix: ""
       bazeliskInstallPath: "/usr/bin/bazel.exe"
       bazeliskReleaseFilename: bazelisk-windows-amd64.exe
       gcrImageName: "envoy-build-windows"
       osFamily: "windows"
+      goPath: "/usr/bin/go.1.16/bin/go"
```
Comment (smukherj1) [inline azure-pipelines.yml:114]: This might need to be `/usr/bin/go.1.16/go/bin/go.exe`

**Comment by yanavlasov** [comment]
Ok, it looks like I finally got it under control.

**Review by sunjayBhatia — DISMISSED** [review]

**Comment by sunjayBhatia** [comment]
@lizan do you want to give this a look for final approval?

File: azure-pipelines.yml [hunk 6]
```diff
@@ -42,13 +42,23 @@ jobs:
   pool:
     vmImage: 'ubuntu-18.04'
   steps:
+  - bash: |
```
Comment (lizan) [inline azure-pipelines.yml:45]: Doesn't the Go Installer task work? https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/go-tool?view=azure-devops

**Review by lizan — APPROVED** [review]

**Pull request merged**